widget_pages/dashboard.py

```diff
@@ -202,5 +280,24 @@ def showPatientListWindow(self):
     def showPatientInformationWindow(self):
         self.parent().parent().showPatientInformationWindow()
 
-    def updatePatientInfo(self):
+    def updatePatientInfo(self, calculation_info):
         self.patient = self.parent().parent().selected_patient
+        if calculation_info[0]:
+            self.graph.toggleResults(True)
+            self.runMatLabModel(self.patient, calculation_info[1])
+            self.displayed_patient = self.patient
+        elif self.displayed_patient and self.displayed_patient.user_id == self.patient.user_id:
+            self.graph.toggleResults(True)
+        else:
+            self.graph.toggleResults(False)
+
+    def runMatLabModel(self, patient, num_cycles):
+        bsa = float(patient.bsa)
+        num_cycles = float(num_cycles + 1)
```

Comment:
why are we running for `num_cycles + 1`?

Reply:
We are assuming that when the user enters 1 cycle to run, they want to run 1 cycle into the future. That means the model needs to run both the current cycle (for reference trajectory) and the future cycles -> 2 cycles to run

Reply:
Actually, I have more questions. If the patient has 3 measurements to use and you specify 2 cycles, does it calculate 5 cycles total? (3 for historical then 2 for the future?)

Reply:
no, it should only run 3 cycles, the current one and 2 into the future

Reply:
So it only runs using the most recent measurement then?

Reply:
yeah, I believe so